sympy/combinatorics/perm_groups.py

```diff
@@ -3230,6 +3231,103 @@ def is_cyclic(self):
         self._is_abelian = True
         return True
 
+    @property
+    def is_dihedral(self):
+        r"""
+        Return ``True`` if the group is dihedral.
+
+        Examples
+        ========
+
+        >>> from sympy.combinatorics.perm_groups import PermutationGroup
+        >>> from sympy.combinatorics.permutations import Permutation
+        >>> from sympy.combinatorics.named_groups import SymmetricGroup, CyclicGroup
+        >>> G = PermutationGroup(Permutation(1, 6)(2, 5)(3, 4), Permutation(0, 1, 2, 3, 4, 5, 6))
+        >>> G.is_dihedral
+        True
+        >>> G = SymmetricGroup(3)
+        >>> G.is_dihedral
+        True
+        >>> G = CyclicGroup(6)
+        >>> G.is_dihedral
+        False
+
+        References
+        ==========
+
+        .. [Di1] https://math.stackexchange.com/a/827273
+        .. [Di2] https://kconrad.math.uconn.edu/blurbs/grouptheory/dihedral.pdf
+        .. [Di3] https://kconrad.math.uconn.edu/blurbs/grouptheory/dihedral2.pdf
+        .. [Di4] https://en.wikipedia.org/wiki/Dihedral_group
+        """
+        if self._is_dihedral is not None:
+            return self._is_dihedral
+
+        order = self.order()
+
+        if order % 2 == 1:
+            self._is_dihedral = False
+            return False
```

Comment:
If we are at Py 3.8 minimum, then is this a place where `return self._is_dihedral := False` could be used?

Reply:
I'm entirely in favor of the walrus, and I went to go try this, but disappointingly it looks like it's not going to work.
I tried

    return (self._is_dihedral := False)

(the parentheses seem to be required, when it does work) but got

    SyntaxError: cannot use assignment expressions with attribute

So it seems it can't be used with attributes. Some explanation is offered at https://stackoverflow.com/a/60909096
Too bad.

Reply:
Thanks for checking and updating.